### PR TITLE
Natasha/messaging ui

### PR DESCRIPTION
**Pull request by Natasha08**
### Description:
[Issue #1019](https://github.com/OriginProtocol/origin/issues/1019)
[Issue #815](https://github.com/OriginProtocol/origin/issues/815)
[Issue #689](https://github.com/OriginProtocol/origin/issues/689)

- update the messaging ui to match comps (mobile and desktop)
- display mobile messaging for smaller screens (max-width: 991px)
- larger text input
- padding/css fixes
- link avatars in conversations to user profiles

### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything
- [x] Wrap any new text/strings for translation
- [ ] Map any new environment variables with a default value in the Webpack config
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/origin-docs)

### Updated Screenshots:
image
image

### Screenshots:
image

image

### CODE REVIEW

**Comment by Natasha08** [comment]
@micahalcorn yes it does -- I'm pushing the textarea change shortly

**Comment by micahalcorn** [comment]
@Natasha08 can we add some padding to the bottom of the bubble? It is especially problematic when images are sent.

File: origin-dapp/src/utils/user.js [hunk 1]
```diff
@@ -13,3 +15,23 @@ const { web3 } = origin.contractService
...
+    + separator
+    + address.substr(address.length - backChars)
+}
+
+export function abbreviateName(party) {
+  const { profile = {}, fullName } = party
+  const { firstName = '', lastName = '' } = profile
+  const lastNameLetter = lastName.length && `${lastName.charAt(0)}.`
+
+  return fullName && `${firstName} ${lastNameLetter}`
```
Comment (micahalcorn) [inline origin-dapp/src/utils/user.js]: @Natasha08 I think that we will get `false` or `0` in the name if one of those values is `undefined`.

Comment (Natasha08) [reply origin-dapp/src/utils/user.js]: @micahalcorn I have tested with an account with no names, but I will double check

**Comment by Natasha08** [comment]
@micahalcorn I’ll take a look at small screens and issue #689 first thing

**Comment by micahalcorn** [comment]
@Natasha08 we seem to have lost the auto-scrolling that should occur when a new message is posted in an existing conversation. This is a problem because the user may not know to scroll and think that the message disappeared.

**Comment by micahalcorn** [comment]
@Natasha08 also getting `Warning: validateDOMNesting(...): <div> cannot appear as a descendant of <p>.`, which may be a simple fix with a `span` or something.

File: origin-dapp/src/components/conversation.js [hunk 2]
```diff
@@ -233,78 +275,81 @@ class Conversation extends Component {
...
+    const user = users.find((user) => formattedAddress(user.address) === partyAddress)
+    const userName = abbreviateName(user)
+    const party = userName || truncateAddress(returnValues.party)
+
+    const offerMessages = {
+      'OfferCreated': `${party} ${intl.formatMessage(
+        this.intlMessages.offerCreated
+      )}`,
+      'OfferWithdrawn': `${party} ${intl.formatMessage(
+        this.intlMessages.offerWithdrawn
```
Comment (micahalcorn) [inline origin-dapp/src/components/conversation.js]: @Natasha08 these should be complete sentences, if possible. I know that we don't always do this elseqhere in the code, but we should ([like this](https://github.com/OriginProtocol/origin/commit/9ed6ead33672b2ea6c46b2b73e89926dcb54cf80#diff-61c348e20f66a533d512610bffbebad3R83)).

File: origin-dapp/src/components/conversation.js [hunk 3]
```diff
@@ -25,14 +26,44 @@ class Conversation extends Component {
...
         defaultMessage: 'Type something...'
-      }
+      },
+      offerCreated: {
+        id: 'conversation.offerCreated',
+        defaultMessage: 'made an offer on'
+      },
+      offerWithdrawn: {
+        id: 'conversation.offerWithdrawn',
+        defaultMessage: 'withdrew their offer on'
```
Comment (micahalcorn) [inline origin-dapp/src/components/conversation.js]: @Natasha08 these should be complete sentences, if possible. I know that we don't always do this elsewhere in the code, but we should ([like this](https://github.com/OriginProtocol/origin/commit/9ed6ead33672b2ea6c46b2b73e89926dcb54cf80#diff-61c348e20f66a533d512610bffbebad3R83)).

File: origin-dapp/src/components/conversation.js [hunk 4]
```diff
@@ -233,78 +275,81 @@ class Conversation extends Component {
...
+    const partyAddress = formattedAddress(returnValues.party)
+    const user = users.find((user) => formattedAddress(user.address) === partyAddress)
+    const userName = abbreviateName(user)
+    const party = userName || truncateAddress(returnValues.party)
+
+    const offerMessages = {
+      'OfferCreated': `${party} ${intl.formatMessage(
+        this.intlMessages.offerCreated
+      )}`,
+      'OfferWithdrawn': `${party} ${intl.formatMessage(
```
Comment (micahalcorn) [inline origin-dapp/src/components/conversation.js]: @Natasha08 this needs to handle the rejection/withdrawal distinction. They are the same at the contract level, so we have to interpret the perspective. See [here](https://github.com/OriginProtocol/origin/commit/9db3ce08a1d231ae922882e66fd36adb88321743#diff-77a57a9603c69e6c5d5af8d7773c0d7eR58).

File: origin-dapp/src/components/conversation.js [hunk 5]
```diff
@@ -233,78 +275,81 @@ class Conversation extends Component {
...
+        this.intlMessages.offerFinalized
+      )}`,
+      'OfferData': `${party} ${intl.formatMessage(
+        this.intlMessages.offerData
+      )}`,
+    }
+
+    return (
+      <div key={new Date() + Math.random()} className="purchase-info">
+        {offerMessages[event]} {listing.name} on {formatDate(timestamp)}
```
Comment (micahalcorn) [inline origin-dapp/src/components/conversation.js]: @Natasha08 while you're here, we should probably go ahead and linkify this message to point to purchase-detail. I'm not too concerned about the styling.

**Comment by micahalcorn** [comment]
Looking great 👏

File: origin-dapp/src/components/conversation.js (hunk 4 above)
Comment (Natasha08) [reply origin-dapp/src/components/conversation.js]: @micahalcorn Okay I'll take a look and update the code

**Comment by Natasha08** [comment]
@micahalcorn not sure how I broke the auto-scrolling but I think I fixed it and addressed your other comments

**Pull request merged**